File: BACK-END/src/main/java/org/kuenteco/backend/dto/profile/UpdateProfileDTO.java
package org.kuenteco.backend.dto.profile;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kuenteco.backend.dto.image.ImageDTO;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UpdateProfileDTO {
    private Integer id;
    private String username;
    private String email;
    private String password;
    private ImageDTO image;
}

File: FRONT-END/lib/dto/app/profile/update_profile_dto.dart
import '../image/image_dto.dart';

class UpdateProfileDTO {
  final int id;
  final String username;
  final String email;
  final ImageDTO? image;
  final bool removeImage;

  UpdateProfileDTO({
    required this.id,
    required this.username,
    required this.email,
    this.image,
    required this.removeImage,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'username': username,
      'email': email,
      'removeImage': removeImage,
    };
  }
}